### DISCUSSION TITLE
Apply GLSL shader to only some letters

### DISCUSSION THREAD

**Discussion by csimlinger** [protectwise/troika#316] [Q&A]
Hi,

I was wondering what solution would you recommend to apply a shader to only some letters?
So far I was generating each letter on a text mesh, so that I could apply a shader to individuell letters, but that was painfully slow, especially for longer text.

What approach is recommended?

Thanks!

**Comment by lojjic** [comment]
(Moved to a discussion)

I can't think of a good built-in way to do that; the nature of instanced rendering is that you can only have one material/shader applied to the whole thing.

If I needed to do that, I might try customizing the shader to apply the special effects to only certain characters, distinguishing them by `gl_InstanceId` (assuming webgl2).

Other ideas, anyone?
